test/pipelinefinally_test.go

```diff
@@ -477,12 +556,17 @@ func getPipeline(t *testing.T, namespace string, dag map[string]pipelineTask, f
 		pt = append(pt, task)
 	}
 	for k, v := range f {
-		fpt = append(fpt, v1beta1.PipelineTask{
-			Name: k,
-
+		finalTask := v1beta1.PipelineTask{
+			Name:    k,
 			TaskRef: &v1beta1.TaskRef{Name: v.TaskName},
-			Params:  v.Param,
-		})
+		}
+		if len(v.Param) != 0 {
+			finalTask.Params = v.Param
+		}
+		if len(v.When) != 0 {
+			finalTask.WhenExpressions = v.When
+		}
+		fpt = append(fpt, finalTask)
```

Comment:
i wonder how useful this function actually is vs declaring a pipeline inline

Reply:
@pritidesai what do you think? maybe we can refactor this in a future PR?

Reply:
sure we can in a separate PR. This function is useful since its shared between two separate tests. We had the tests implemented without this common function and found a lot of repetition. We can declare a pipeline inline in favor of the readability instead of using this function.